src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -2,21 +2,60 @@
 
 import java.util.Optional;
 import mate.academy.hibernate.relations.dao.ActorDao;
+import mate.academy.hibernate.relations.exeption.DataProcessingException;
 import mate.academy.hibernate.relations.model.Actor;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 
 public class ActorDaoImpl extends AbstractDao implements ActorDao {
+
+    public static final String CAN_T_ADD_ACTOR_TO_DB_MSG = "Can't add actor to db: ";
+    public static final String CAN_T_GET_ACTOR_BY_ID_MSG = "Can't get actor by id: ";
+
     public ActorDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(CAN_T_ADD_ACTOR_TO_DB_MSG + actor,e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return actor;
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        Transaction transaction = null;
+        try (Session session = factory.openSession()) {
+            transaction = session.beginTransaction();
+            Actor actor = session.get(Actor.class, id);
+            transaction.commit();
```

Comment:
The transaction management in the `get` method is unnecessary because reading operations do not require a transaction unless you are concerned about repeatable reads or other transaction isolation issues. You can remove the transaction management for a simple read operation like this.